nsqd/nsqd.go

```diff
@@ -85,32 +81,13 @@ func NewNSQD(opts *nsqdOptions) *NSQD {
 		os.Exit(1)
 	}
 
-	tcpAddr, err := net.ResolveTCPAddr("tcp", opts.TCPAddress)
-	if err != nil {
-		n.logf("FATAL: failed to resolve TCP address (%s) - %s", opts.TCPAddress, err)
-		os.Exit(1)
-	}
-	n.tcpAddr = tcpAddr
-
-	httpAddr, err := net.ResolveTCPAddr("tcp", opts.HTTPAddress)
-	if err != nil {
-		n.logf("FATAL: failed to resolve HTTP address (%s) - %s", opts.HTTPAddress, err)
-		os.Exit(1)
-	}
-	n.httpAddr = httpAddr
-
-	if opts.HTTPSAddress != "" {
-		httpsAddr, err := net.ResolveTCPAddr("tcp", opts.HTTPSAddress)
+	if opts.StatsdPrefix != "" {
+		_, port, err := net.SplitHostPort(opts.HTTPAddress)
```

Comment:
looks like `opts.HTTPAddress` got rolled into the `statsdPrefix` block accidentally